data/modules/TradeShips/Trader.lua

```diff
@@ -0,0 +1,355 @@
+	if ship_type.equipSlotCapacity.ecm > 0 then
+		if Engine.rand:Number(1) + 0.2 < lawlessness then
+			ship:AddEquip(e.misc.ecm_advanced)
+		elseif Engine.rand:Number(1) < lawlessness then
+			ship:AddEquip(e.misc.ecm_basic)
+		end
+	end
+
+	-- this should be rare
+	if ship_type.equipSlotCapacity.hull_autorepair > 0 and
+		Engine.rand:Number(1) + 0.75 - size_factor < lawlessness then
+		ship:AddEquip(e.misc.hull_autorepair)
+	end
+end
+
+Trader.addCargo = function (ship, direction)
+	local imports = Core.params.imports
+	local exports = Core.params.exports
+	local total = 0
+	local empty_space = math.min(ship.freeCapacity, ship:GetEquipFree("cargo"))
+	local size_factor = empty_space / 20
+	local ship_cargo = {}
+
+	if direction == 'import' and #imports == 1 then
+		total = ship:AddEquip(imports[1], empty_space)
+		ship_cargo[imports[1]] = total
+	elseif direction == 'export' and #exports == 1 then
+		total = ship:AddEquip(exports[1], empty_space)
+		ship_cargo[exports[1]] = total
+	elseif (direction == 'import' and #imports > 1) or
+		(direction == 'export' and #exports > 1) then
+
+		-- happens if there was very little space left to begin with (eg small
+		-- ship with lots of equipment). if we let it through then we end up
+		-- trying to add 0 cargo forever
+		if size_factor < 1 then
+			if Core.ships[ship] then
+				Core.ships[ship]['cargo'] = ship_cargo
+			end
+			return 0
+		end
+
+		while total < empty_space do
+			local cargo_type
+
+			-- get random for direction
+			if direction == 'import' then
+				cargo_type = imports[Engine.rand:Integer(1, #imports)]
+			else
+				cargo_type = exports[Engine.rand:Integer(1, #exports)]
+			end
+
+			-- amount based on price and size of ship
+			local num = math.abs(Game.system:GetCommodityBasePriceAlterations(cargo_type.name)) * size_factor
+			num = Engine.rand:Integer(num, num * 2)
+
+			local added = ship:AddEquip(cargo_type, num)
+			if ship_cargo[cargo_type] == nil then
+				ship_cargo[cargo_type] = added
+			else
+				ship_cargo[cargo_type] = ship_cargo[cargo_type] + added
+			end
+			total = total + added
+		end
+	end
+	if Core.ships[ship] then
+		Core.ships[ship]['cargo'] = ship_cargo
+	end
+	-- if the table for direction was empty then cargo is empty and total is 0
+	return total
+end
+
+Trader.doOrbit = function (ship)
+	local trader = Core.ships[ship]
+	local sbody = trader.starport.path:GetSystemBody()
+	local body = Space.GetBody(sbody.parent.index)
+	ship:AIEnterLowOrbit(body)
+	trader['status'] = 'orbit'
+	Core.log:add(ship, 'Ordering orbit of '..body.label)
+end
+
+local getSystem = function (ship)
+	local max_range = ship:GetEquip('engine', 1):GetMaximumRange(ship)
+	if max_range > 30 then
+		max_range = 30
+	end
+	local min_range = max_range / 2;
+	local systems_in_range = Game.system:GetNearbySystems(min_range)
+	if #systems_in_range == 0 then
+		systems_in_range = Game.system:GetNearbySystems(max_range)
+	end
+	if #systems_in_range == 0 then
+	return nil end
```

Comment:
This line should either be indented or on the previous line following `then`.

Reply:
Done!